### PR TITLE
GEOMESA-2166 Adding support for Cloudera HBase 1.2

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala [hunk 1]
```diff
@@ -79,10 +80,7 @@ trait HBaseFeatureIndex extends HBaseFeatureIndexType with ClientSideFiltering[R
       if (!desc.getCoprocessors.contains(name)) {
         // TODO: Warn if the path given is different from paths registered in other coprocessors
         // if so, other tables would need updating
-        coprocessorUrl match {
-          case Some(path) => desc.addCoprocessor(name, path, Coprocessor.PRIORITY_USER, null)
```
Comment (aheyne) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala]: Do we still need this check? Not sure what the reflection is going to do if you don't have a url/path.

Comment (elahrvivaz) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala]: I moved it to the API shim - it takes an `Option[Path]` and uses default values that correspond to the default values in `HTableDescriptor.addCoprocessor(String)` (which calls `HTableDescriptor.addCoprocessor(String, Path, Int, Map) with PRIORITY_USER and nulls)`.

Comment (elahrvivaz) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala]: https://github.com/elahrvivaz/geomesa/blob/13f23f98524f0138b1c2da7e9919002d2ea334cb/geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/HBaseVersions.scala#L25

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/HBaseVersions.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.hbase.utils
+
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.hbase.{Coprocessor, HColumnDescriptor, HTableDescriptor}
+
+/**
+  * Reflection wrapper for AdminUtils methods between kafka versions 0.9 and 0.10
```
Comment (jahhulbert-ccri) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/HBaseVersions.scala]: kafka?

Comment (elahrvivaz) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/HBaseVersions.scala]: oops, copy paste fail

**Comment by elahrvivaz** [comment]
@jahhulbert-ccri cloudera has some custom hbase jars that have slightly different methods than the normal ones. This causes `NoSuchMethodError`s to get thrown. So I've added some reflection wrappers that get around the differences in method signature.

**Comment by jahhulbert-ccri** [comment]
ewwwwwww cloudera :)

**Pull request merged**